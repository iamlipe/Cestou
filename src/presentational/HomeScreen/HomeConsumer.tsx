import React from 'react';
import styled from 'styled-components/native';
import i18next from 'i18next';
import {useTranslation} from 'react-i18next';

import imgBanner from '@/assets/images/presentation_1.png';
import imgOngOne from '@/assets/images/ong_1.png';
import imgOngTwo from '@/assets/images/ong_2.png';

import Header from '@/components/Header';
import CardOng from '@/components/CardOng';

interface TextProps {
  title?: boolean;
}

interface DataOng {
  image: string;
  title: string;
  link: string;
}

const dataOngServices: DataOng[] = [
  {
    image: imgOngOne,
    title: i18next.t('CardOng.One'),
    link: 'https://cestou.netlify.app/sobre',
  },
  {
    image: imgOngTwo,
    title: i18next.t('CardOng.Two'),
    link: 'https://cestou.netlify.app/sobre',
  },
];

export const HomeConsumer = () => {
  const {t} = useTranslation();

  return (
    <StyledContainerScroll showsVerticalScrollIndicator={false}>
      <Header />
      <StyledBanner source={imgBanner}>
        <StyledTitleBanner>
          {t('Text.ScreenHomeConsumer.TitleBanner')}
        </StyledTitleBanner>
        <StyledSubtitleBanner>
          {t('Text.ScreenHomeConsumer.SubTitleBanner')}
        </StyledSubtitleBanner>
      </StyledBanner>
      <StyledContent>
        <StyledText title>{t('Text.ScreenHomeConsumer.Title')}</StyledText>
        <StyledText>{t('Text.ScreenHomeConsumer.SubTitle')}</StyledText>

        <StyledContainerServices>
          {dataOngServices.map((ong: DataOng, index: number) => (
            <CardOng
              key={index}
              title={ong.title}
              image={ong.image}
              link={ong.link}
            />
          ))}
        </StyledContainerServices>
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledContainerScroll = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

const StyledContent = styled.View`
  width: 90%;
  align-self: center;
  padding: 24px 0;
`;

const StyledBanner = styled.ImageBackground`
  width: 100%;
  height: 198px;
  align-items: flex-end;

  padding: 16px;
`;

const StyledTitleBanner = styled.Text`
  font-style: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.LARGEST};
  color: ${({theme}) => theme.colors.PRIMARY_800};

  text-align: right;
  margin-bottom: 16px;
`;

const StyledSubtitleBanner = styled(StyledTitleBanner)`
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
  line-height: 22px;
`;

const StyledText = styled.Text<TextProps>`
  font-family: ${({theme, title}) =>
    title
      ? theme.fonts.SEMIBOLD_SOURCESANSPRO
      : theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme, title}) =>
    title ? theme.sizing.SMALL : theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};

  margin-bottom: 16px;
`;

const StyledContainerServices = styled.View`
  flex-direction: row;
  justify-content: space-between;
`;
