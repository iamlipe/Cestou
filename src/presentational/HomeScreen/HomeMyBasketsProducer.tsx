import React, {useEffect, useState} from 'react';
import styled from 'styled-components/native';
import * as Yup from 'yup';
import {yupResolver} from '@hookform/resolvers/yup';
import {SvgProps} from 'react-native-svg';
import {useForm} from 'react-hook-form';
import {useReduxDispatch} from '@/hooks/useReduxDispatch';
import {useReduxSelector} from '@/hooks/useReduxSelector';
import {useTranslation} from 'react-i18next';
import {useNavigation} from '@react-navigation/native';
import {
  GET_BASKET,
  SIGNUP_PRODUCER_BASKET,
  BasketResponse,
} from '@/store/slices/basketSlice';

import {translateBasketToEnglish} from '@/helpers/translate';

import IconVegetable from '@/assets/svgs/vegetable.svg';

import Header from '@/components/Header';
import Checkbox from '@/components/CheckboxForm';
import Button from '@/components/Button';
import Modal from '@/components/Modal';

interface SignupBasket {
  myBasket: string[];
}

export const HomeMyBasketsProducer = () => {
  const [isVisibleModal, setIsVisibleModal] = useState(false);
  const {isLoading, allBaskets} = useReduxSelector(state => state.basket);
  const {goBack} = useNavigation();
  const {t} = useTranslation();
  const dispatch = useReduxDispatch();

  const schema = Yup.object().shape({
    myBasket: Yup.array()
      .min(1, t('error.minProduct'))
      .required(t('error.required')),
  });

  const {
    control,
    handleSubmit,
    formState: {errors, isSubmitted},
  } = useForm<SignupBasket>({
    resolver: yupResolver(schema),
  });

  function handleModal(event: boolean) {
    setIsVisibleModal(event);
  }

  function signProducerBasket(data: SignupBasket) {
    data.myBasket.forEach((selectedBasket: string) => {
      const translatedBasket = translateBasketToEnglish(selectedBasket);

      const basketID = allBaskets.find(
        (basket: BasketResponse) => basket.size === translatedBasket,
      )?.id;

      if (basketID) dispatch(SIGNUP_PRODUCER_BASKET({basketID}));
    });
  }

  async function onSubmit(data: SignupBasket) {
    signProducerBasket(data);
    handleModal(true);
  }

  useEffect(() => {
    dispatch(GET_BASKET());
  }, [dispatch]);

  return (
    <StyledContainerScroll showsVerticalScrollIndicator={false}>
      <Header
        title={t('text.screenHomeMyBasketsProducer.titleHeader')}
        welcome={false}
      />
      <StyledContent>
        <StyledText>
          {t('text.screenHomeMyBasketsProducer.instructions')}
        </StyledText>
        <StyledObservationText>
          {t('text.screenHomeMyBasketsProducer.note')}
        </StyledObservationText>

        <Checkbox
          name="myBasket"
          control={control}
          options={[
            t('option.smallBasket'),
            t('option.mediumBasket'),
            t('option.largeBasket'),
          ]}
          detailsOptions={[
            t('detailsOption.smallBasket'),
            t('detailsOption.mediumBasket'),
            t('detailsOption.largeBasket'),
          ]}
          error={isSubmitted ? errors.myBasket?.message : ''}
        />

        <StyledSubmitButton
          testID="subimit-button"
          title={t('button.confirm')}
          loading={isLoading}
          onPress={handleSubmit(onSubmit)}
        />

        {!isLoading && isVisibleModal && (
          <Modal
            title={t('modal.titleConfirmProducerBasket')}
            icon={IconVegetable as React.FC<SvgProps>}
            onConfirm={() => {
              handleModal(false);
              goBack();
            }}
            onCancel={() => handleModal(false)}
            onClose={() => handleModal(false)}
          />
        )}
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledContainerScroll = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

const StyledContent = styled.View`
  width: 90%;
  align-self: center;
  padding: 24px 0;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};

  margin-bottom: 16px;
`;

const StyledObservationText = styled(StyledText)`
  font-size: ${({theme}) => theme.sizing.SMALLEST};
`;

const StyledSubmitButton = styled(Button)`
  margin-top: 24px;
`;
