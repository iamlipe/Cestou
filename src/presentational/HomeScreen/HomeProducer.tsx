import React from 'react';
import styled from 'styled-components/native';
import {t} from 'i18next';
import {useNavigation} from '@react-navigation/native';
import {HomeProducerStackParamList} from '@/routes/stacks/HomeProducerStack';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';

import Header from '@/components/Header';
import Button from '@/components/Button';

type NavProps = NativeStackNavigationProp<
  HomeProducerStackParamList,
  'HomeMyBasketsProducer'
>;

export const HomeProducer = () => {
  const {navigate} = useNavigation<NavProps>();

  const renderLargeButton = (title: string) => (
    <StyledContainerLargeButton>
      <StyledTitleLargeButton>{title}</StyledTitleLargeButton>
    </StyledContainerLargeButton>
  );

  return (
    <StyledContainer showsVerticalScrollIndicator={false}>
      <Header />
      <StyledTitle>{t('text.screenHomeProducer.titleSectionOne')}</StyledTitle>
      <StyledContainerAboutMyBasket>
        <StyledTitleAboutMyBasket>
          {t('text.screenHomeProducer.note')}
        </StyledTitleAboutMyBasket>

        <ButtonAboutMyBasket
          title={t('button.myBaskets')}
          onPress={() => navigate('HomeMyBasketsProducer')}
        />
      </StyledContainerAboutMyBasket>

      <StyledTitle>{t('text.screenHomeProducer.titleSectionTwo')}</StyledTitle>
      {renderLargeButton(t('button.myEarnings'))}
      {renderLargeButton(t('button.registerPix'))}
    </StyledContainer>
  );
};

const StyledContainer = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

const StyledTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALL};
  color: ${({theme}) => theme.colors.GRAY_900};
  margin: 24px 5% 16px;
`;

const StyledContainerAboutMyBasket = styled.View`
  width: 90%;
  height: 130px;
  align-self: center;
  align-items: center;
  justify-content: space-evenly;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
  border-radius: 4px;
  border: 1px solid ${({theme}) => theme.colors.GRAY_100};

  padding: 0 8px;
`;

const StyledTitleAboutMyBasket = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLEST};
  color: ${({theme}) => theme.colors.GRAY_900};

  margin-bottom: 8px;
`;

const ButtonAboutMyBasket = styled(Button)``;

const StyledContainerLargeButton = styled.TouchableOpacity`
  width: 90%;
  height: 100px;
  align-self: center;
  justify-content: center;
  align-items: center;
  background-color: ${({theme}) => theme.colors.PRIMARY_100};
  border-radius: 4px;
  border: 1px solid ${({theme}) => theme.colors.PRIMARY_300};
  margin-bottom: 16px;
`;

const StyledTitleLargeButton = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
`;
