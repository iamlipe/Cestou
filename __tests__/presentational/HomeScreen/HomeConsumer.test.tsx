import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import {HomeConsumer} from '@/presentational/HomeScreen/HomeConsumer';

describe('HomeConsumer', () => {
  test('should render screen correctly', () => {
    renderWithThemeProvider(<HomeConsumer />);
  });
  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<HomeConsumer />);
  });
});
