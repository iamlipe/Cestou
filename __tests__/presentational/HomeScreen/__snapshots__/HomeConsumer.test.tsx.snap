// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeConsumer should matches snapshot 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "minHeight": "100%",
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6CCD91",
              "flexDirection": "row",
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": "5%",
              "paddingRight": "5%",
              "paddingTop": 0,
              "width": "100%",
            },
          ]
        }
        type="normal"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "space-between",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1A1A1A",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 47,
                  "marginLeft": 10,
                },
              ]
            }
          >
            Olá !
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "height": 198,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/presentation_1.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 198,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#00843F",
              "fontSize": 47,
              "fontStyle": "SourceSansPro-SemiBold",
              "marginBottom": 16,
              "textAlign": "right",
            },
          ]
        }
      >
        Faça sua
 assinatura!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontSize": 27,
              "fontStyle": "SourceSansPro-SemiBold",
              "lineHeight": 22,
              "marginBottom": 16,
              "textAlign": "right",
            },
          ]
        }
      >
        Produtos fresquinhos de
 produtores locais toda
 semana na sua mesa
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingBottom": 24,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 24,
            "width": "90%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 31,
              "marginBottom": 16,
            },
          ]
        }
        title={true}
      >
        Ajude no combate a fome
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 27,
              "marginBottom": 16,
            },
          ]
        }
      >
        Veja abaixo as instituições e projetos cadastrados na nossa rede e ajude-nos a atingir a meta do mês doando suas moedas que serão convertidas em alimentos.
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "width": "48%",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/ong_1.png",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 4,
                  "height": 130,
                  "marginBottom": 5,
                  "width": "100%",
                },
              ]
            }
            testID="card-ong-image"
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#1A1A1A",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 27,
                },
              ]
            }
          >
            ONG Comida na mesa
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "width": "48%",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/ong_2.png",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 4,
                  "height": 130,
                  "marginBottom": 5,
                  "width": "100%",
                },
              ]
            }
            testID="card-ong-image"
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#1A1A1A",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 27,
                },
              ]
            }
          >
            ONG Criança sem fome
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
